Maps mounted data dependencies to normalizer packages

bwalsh opened the pull request:
This PR adds a docker file that includes a script `discover_normalizer_paths.sh`  that discovers the pipenv paths and maps mounted directories to package paths.


```
     - ./data/therapy:/app/therapy/data          
     - ./data/gene:/app/gene/data
     - ./data/disease:/app/disease/data
```

OHSU-MachineUser committed:
Maps mounted data dependencies to normalizer packages

richardhj commented:
```Dockerfile
RUN chmod +x discover_normalizer_paths.sh
```
needs to be added, otherwise, you may run into a "command not found in pipenv path" error.

zflohr committed:
Merge branch 'main' into docker

zflohr committed:
Merge branch 'main' into docker

zflohr committed:
Update names of envars storing normalizer paths in shell script

zflohr committed:
Fix pipenv commands and lock development packages in Pipfile.lock

zflohr committed:
Add .dockerignore

zflohr committed:
Add comments to Dockerfile

zflohr committed:
Add comments to Dockerfile, and handle TSV file for disease service

- Ensure that TSV file from omim.org is placed in the correct directory
  in the disease-normalizer PyPI package in the site-packages directory
  of the metakb virtual environment in the container.

zflohr committed:
Replace disease-normalizer etl.ncit.py with another file in container

- The file etl.ncit.py that comes with disease-normalizer v0.2.15
  contains a variable that stores an incorrect file path on the NCIT
  server. This file must be replaced with another file that has been
  hardcoded to include the correct file path. The replacement occurs per
  removal of the original file in the container and a symbolic link to a
  bind-mounted file.

zflohr committed:
Pin chembl-downloader to a lower version in Pipfile

- chembl-downloader v0.4.2 is a subdependency that gets locked in
  Pipfile.lock. This version causes errors during ETL of ChEMBL data.
  Pin package to v0.4.1 in Pipfile.

korikuzma commented:
Going to close since this is an old PR.
